Create CSS Entry for floats (#618)
Codecademy/docs: content/css/concepts/floats/floats.md

---
Title: 'Floats'
Description: ''
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Floats'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
---

The `float` propertie moves an element with in its parent.

## Syntax

> The `float` property takes 5 different values. They are, `left`, `right`, and `none`. The value `left` makes the element float on the left of a parent. The value `right` makes the element float on the right of a parent. The value `none` makes the element not float at all.

Reviewer: Here you have added 5 values and you have described only 3, could you please add other 2 as well i.e. initial and inherit? with description?

---
Title: 'Floats'
Description: 'The float property moves an element to the left or right within its parent, outside of the default positioning.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Floats'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
---

The float property moves an element to the left or right within its parent, outside of the default positioning.

## Syntax

> 

Reviewer 2: For the "Syntax" it is preferable to include a \```pseudo code snippet that visualizes the values you're describing here.

> The `float` property takes 3 different values. They are, `left`, `right`, and `none`. The value `left` makes the element float on the left of a parent. The value `right` makes the element float on the right of a parent. The floating element will render beside elements with the default positioning. The value `none` removes the floating property from an element.

Reviewer 2: ```suggestion
The `float` property primarily uses the following three values:

* The `left` value that makes the element float on the left of its container.
* The `right` value that makes the element float on the right of its container.
* The `none` value that causes the element not to float.
```

## Example 1

```html
<section>
  <div class="item_1">
    <p>This is the first item</p>
  </div>
  <div class="item_2">
    <p>This is the second item</p>
  </div>
  <p>This example shows CSS floats.</p>
</section>
```

> 

Reviewer 2: Add 1-2 transition sentences that talk about the setup in the previous \```html snippet and how they can be selected in a separate stylesheet with class selectors (`.item_x`, etc.).

```css
.item_1 {
  float: left;
}

.item_2 {
  float: right;
}
```

> The div with the class `item_1` will float to the left of the text `This example shows CSS floats.` And the div with the class `item_2` will float to the right of the text `This example shows CSS floats.`

Reviewer 2: Can include a supporting image for this in `media/`?

> The div with the class `item_1` will float to the left of the text `This example shows CSS floats.` And the div with the class `item_2` will float to the right of the text `This example shows CSS floats.`

Reviewer 2: Can include a supporting image for this in `media/`?

> The div with the class `item_1` will float to the left of the text `This example shows CSS floats.` And the div with the class `item_2` will float to the right of the text `This example shows CSS floats.`

Reviewer 2: ```suggestion
The `<div>` tags with the `.item_1` and `.item_2` classes will float to the left and right of the "CSS floats" text, respectively.
```

---
Title: 'Floats'
Description: 'The float property moves an element to the left or right within its parent, outside of the default positioning.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Floats'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
---

The `float` property moves an element to the left or right within its containing element, outside of the default [positioning](https://www.codecademy.com/resources/docs/css/position).

## Syntax

The `float` property primarily uses the following three values:

* The `left` value that makes the element float on the left of its container.
* The `right` value that makes the element float on the right of its container.
* The `none` value that causes the element not to float.

```pseudo
.item_x {
  float: none;
}

.item_y {
  float: left;
}

.item_z {
  float: right;
}
```

## Example

```html

> <section>
> <div class="item_1">
>   <p>This is the first item and I will float left.</p>
> </div>
> <div class="item_2">
>   <p>This is the second item and I will float right</p>
> </div>
>   <br>
>   <br>
>   <br>
> <p>This example shows CSS floats and I wont float at all.</p>

Reviewer 2: ```suggestion
<section>
  <div class="item_1">
    <p>This is the first item and I will float left.</p>
  </div>
  <div class="item_2">
    <p>This is the second item and I will float right</p>
  </div>
    <br>
    <br>
    <br>
  <p>This example shows CSS floats and I wont float at all.</p>
</section>
```

Noticed a missing closing `</section>`

---
Title: 'Floats'
Description: 'The float property moves an element to the left or right within its parent, outside of the default positioning.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Floats'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
---

The `float` property moves an element to the left or right within its containing element, outside of the default [positioning](https://www.codecademy.com/resources/docs/css/position).

## Syntax

The `float` property primarily uses the following three values:

* The `left` value that makes the element float on the left of its container.
* The `right` value that makes the element float on the right of its container.

> * The `none` value that causes the element not to float.
> 

Reviewer 2: ```suggestion
* The `none` value that causes the element not to float.
```

Author: @Dusch4593 There's no change here. What is it supposed to be?

Reviewer 2: Delete line  # 23 (the extra empty line)

```pseudo
.item_x {
  float: none;
}

.item_y {
  float: left;
}

.item_z {
  float: right;
}
```

## Example

```html

>   <section>
>   <div class="item_1">
>     <p>This is the first item and I will float left.</p>
>   </div>
>   <div class="item_2">
>     <p>This is the second item and I will float right</p>
>   </div>
>     <br>
>     <br>
>     <br>
>   <p>This example shows CSS floats and I wont float at all.</p>

Reviewer 2: ```suggestion
<section>
<div class="item_1">
  <p>This is the first item and I will float left.</p>
</div>
<div class="item_2">
  <p>This is the second item and I will float right</p>
</div>
  <br>
  <br>
  <br>
<p>This example shows CSS floats and I wont float at all.</p>
```

```

In the above snipit the elements with the class `item_1` and `item_2` you can set the float in an external style sheet.

```css
.item_1 {
  float: left;
}

.item_2 {
  float: right;
}

> section > p {
>   border: thick double black;
> }

Reviewer 2: Actually let's keep that `<div>` rule for the borders and add the following additional rule:

```suggestion
div {
  border: thick double black;
}

section > p {
  margin: auto;
  text-align: center;
  width: 50%;
}
```

```

> The `<div>` tags with the `.item_1` and `.item_2` classes will float to the left and right of the "CSS floats" text, respectively.

Reviewer 2: Should we center the "CSS floats" text?

```css
section > p {
  text-align: center;
}
```

Author: @Dusch4593 I cant remember what the `>` sign does. Can you remind me?

Reviewer 2: It selects all `<p>` tags where it's containing parent tag is a `<section>` element.

> ![Result of the above code](https://raw.githubusercontent.com/Codecademy/docs/Floats/media/css_floats.png)

Reviewer 2: ```suggestion
![Image of rendered float elements](https://raw.githubusercontent.com/Codecademy/docs/Floats/media/css_floats.png)
```

---
Title: 'Floats'
Description: 'The float property moves an element to the left or right within its parent, outside of the default positioning.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Floats'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
---

The `float` property moves an element to the left or right within its containing element, outside of the default [positioning](https://www.codecademy.com/resources/docs/css/position).

## Syntax

The `float` property primarily uses the following three values:

- The `left` value that makes the element float on the left of its container.
- The `right` value that makes the element float on the right of its container.
- The `none` value that causes the element not to float.

```pseudo
.item_x {
  float: none;
}

.item_y {
  float: left;
}

.item_z {
  float: right;
}
```

## Example

```html
<section>
  <div class="item_1">
    <p>This is the first item and I will float left.</p>
  </div>
  <div class="item_2">

>     <p>This is the second item and I will float right</p>

Reviewer 3: ```suggestion
    <p>This is the second item and I will float right.</p>
```

  </div>
  <br />
  <br />
  <br />
  <p>This example shows CSS floats and I wont float at all.</p>
</section>
```

> In the above snipit the elements with the class `item_1` and `item_2` you can set the float in an external style sheet.

Reviewer 3: ```suggestion
In the above snippet the elements with the class `item_1` and `item_2` you can set the float in an external style sheet.
```

```css
.item_1 {
  float: left;
}

.item_2 {
  float: right;
}

div {
  border: thick double black;
}

section > p {
  margin: auto;
  text-align: center;
  width: 50%;
}
```

The `<div>` tags with the `.item_1` and `.item_2` classes will float to the left and right of the "CSS floats" text, respectively.

> ![Image of rendered float elements](https://raw.githubusercontent.com/Codecademy/docs/Floats/media/css_floats.png)

Reviewer 2: @ethan-master-coding Actually _super_ last comment/suggestion: Can we replace this image with a new one that reflects the style changes we just added? (with the "CSS Floats" text in the middle)

Reviewer 2: Beautiful!
